src/api.js

```diff
@@ -605,6 +633,96 @@ Module["onRuntimeInitialized"] = function onRuntimeInitialized() {
         return res;
     };
 
+    /**
+     * @classdesc
+     * An iterator over multiple SQL statements in a string,
+     * preparing and returning a Statement object for the next SQL
+     * statement on each iteration.
+     *
+     * You can't instantiate this class directly, you have to use a
+     * {@link Database} object in order to create a statement iterator
+     *
+     * {@see Database#iterateStatements}
+     *
+     * **Warning**: When you close a database (using db.close()),
+     * using any statement iterators created by the database will
+     * result in undefined behavior.
+     *
+     * @example
+     * // loop over and execute statements in string sql
+     * for (let statement of db.iterateStatements(sql) {
+     *     statement.step();
+     *     // get results, etc.
+     *     // do not call statement.free()
+     * }
+     *
+     * // capture any bad query exceptions with feedback
+     * // on the bad sql
+     * let it = db.iterateStatements(sql);
+     * try {
+     *     for (let statement of it) {
+     *         statement.step();
+     *     }
+     * } catch(e) {
+     *     console.log(e);  // error message
+     *     console.log(" occurred while executing:");
+     *     console.log(it.getRemainingSQL());
```

Comment:
```suggestion
     *     console.log(`The SQL string "${it.getRemainingSQL()}" contains the following error: ${e}`);
```

Reply:
The linter is complaining that this line is too long... are comments allowed to exceed the limit?